packages/docprovider/src/tokens.ts

```diff
@@ -13,27 +13,9 @@ export const IDocumentProviderFactory = new Token<IDocumentProviderFactory>(
  */
 export interface IDocumentProvider {
   /**
-   * Resolves to true if the initial content has been initialized on the server. false otherwise.
+   * Returns a Promise that resolves when renaming is ackownledged.
    */
-  requestInitialContent(): Promise<boolean>;
-
-  /**
-   * Put the initialized state.
-   */
-  putInitializedState(): void;
-
-  /**
-   * Acquire a lock.
-   * Returns a Promise that resolves to the lock number.
-   */
-  acquireLock(): Promise<number>;
-
-  /**
-   * Release a lock.
-   *
-   * @param lock The lock to release.
-   */
-  releaseLock(lock: number): void;
+  readonly renameAck: Promise<boolean>;
 
   /**
    * This should be called by the docregistry when the file has been renamed to update the websocket connection url
```

Comment:
When does it make sense for us to move to tracking these things by the file id. That would simplify the logic of renaming signifciantly and also improve the experience when users move/renaming the file in the Terminal.

Reply:
Yes, renaming should go through the file ID service. But we still need to support "HTTP-renaming", right? Or should be "deprecate" it when in collaborative mode?